library/smoothstreaming/src/main/java/com/google/android/exoplayer2/source/smoothstreaming/SsMediaSource.java

```diff
@@ -192,9 +192,9 @@ private SsMediaSource(SsManifest manifest, Uri manifestUri,
       AdaptiveMediaSourceEventListener eventListener) {
     Assertions.checkState(manifest == null || !manifest.isLive);
     this.manifest = manifest;
-    this.manifestUri = manifestUri == null ? null
-        : Util.toLowerInvariant(manifestUri.getLastPathSegment()).equals("manifest") ? manifestUri
-            : Uri.withAppendedPath(manifestUri, "Manifest");
+    String lastPathSegment = Util.toLowerInvariant(manifestUri.getLastPathSegment());
```

Comment:
This will NullPointerException is manifestUri is null.

Reply:
You are right but by pulling together the reg-exp to check also for just 'manifest' we should be able to get rid of that local variable entirely.